permissions/resource_rule_set.go

```diff
@@ -0,0 +1,177 @@
+package permissions
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type RuleSetApi struct {
+	client *common.DatabricksClient
+	ctx    context.Context
+}
+
+func NewRuleSetApi(ctx context.Context, m interface{}) RuleSetApi {
```

Comment:
please don't define `RuleSetApi` in the TF, because we already have it in the SDK.